### PR TITLE
Transaction class added

### PR DESCRIPTION
**Pull request by Razybehbahani**

### CODE REVIEW

**Comment by docrobertson** [comment]
The __pycache__ directory in utils also needs to be removed

**Comment by Razybehbahani** [comment]
> The **pycache** directory in utils also needs to be removed

I modified the .gitignore file to exclude them ( _pycache__ => __pycache__/ ), the .gitignore is also in .gitignore file, so it won't show the latest update. For now, we can delete __pycache__ from utils folder manually and from now on no __pycache__ folder will be pushed.

**Review by ttricco — COMMENTED** [review]
The transaction parts for Person needs to be re-designed. I don't think they should store individual transactions (or parts of a transaction), for example. We should discuss in tomorrow's meeting, along with the general set of methods that belong on a Person.

**Comment by Razybehbahani** [comment]
Great! Discussions are enlightening!

Cheers,
Razyeh

File: bankcraft/agent.py [hunk 1]
```diff
@@ -22,10 +23,38 @@ def __init__(self, unique_id, model,
         self._spendingProb = spending_prob
         self._spendingAmount = spending_amount
         self._salary = salary
+        self._tx_type = None
```
Comment (docrobertson) [inline bankcraft/agent.py:26]: Why does agent have a tx_type? An agent isn't a transaction.

**Comment by ttricco** [comment]
@Razybehbahani We discussed this morning around the code restructure for transactions a Person agent makes. Do you want this PR merged in advance of that, or should this be closed without merging?

**Comment by Razybehbahani** [comment]
Before cleaning the agent.py file

**Pull request closed without merging**